lib/msal-browser/src/interaction_handler/SilentHandler.ts

```diff
@@ -4,266 +4,210 @@
-        authCodeModule: AuthorizationCodeClient,
-        storageImpl: BrowserCacheManager,
-        authCodeRequest: CommonAuthorizationCodeRequest,
-        logger: Logger,
-        systemOptions: Required<
-            Pick<
-                BrowserSystemOptions,
-                "navigateFrameWait" | "pollIntervalMilliseconds"
-            >
-        >,
-        performanceClient: IPerformanceClient
-    ) {
-        super(
-            authCodeModule,
-            storageImpl,
-            authCodeRequest,
+/**
+ * Creates a hidden iframe to given URL using user-requested scopes as an id.
+ * @param urlNavigate
+ * @param userRequestScopes
+ */
+export async function initiateAuthRequest(
+    requestUrl: string,
+    performanceClient: IPerformanceClient,
+    logger: Logger,
+    correlationId: string,
+    navigateFrameWait?: number
+): Promise<HTMLIFrameElement> {
+    performanceClient.addQueueMeasurement(
+        PerformanceEvents.SilentHandlerInitiateAuthRequest,
+        correlationId
+    );
+
+    if (!requestUrl) {
+        // Throw error if request URL is empty.
+        logger.info("Navigate url is empty");
+        throw createBrowserAuthError(BrowserAuthErrorCodes.emptyNavigateUri);
+    }
+    if (navigateFrameWait) {
+        return await invokeAsync(
+            loadFrame,
+            PerformanceEvents.SilentHandlerLoadFrame,
             logger,
-            performanceClient
-        );
-        this.navigateFrameWait = systemOptions.navigateFrameWait;
-        this.pollIntervalMilliseconds = systemOptions.pollIntervalMilliseconds;
+            performanceClient,
+            correlationId
+        )(requestUrl, navigateFrameWait, performanceClient, correlationId);
     }
+    return invoke(
+        loadFrameSync,
+        PerformanceEvents.SilentHandlerLoadFrameSync,
+        logger,
+        performanceClient,
+        correlationId
+    )(requestUrl);
+}
 
-    /**
-     * Creates a hidden iframe to given URL using user-requested scopes as an id.
-     * @param urlNavigate
-     * @param userRequestScopes
-     */
-    async initiateAuthRequest(requestUrl: string): Promise<HTMLIFrameElement> {
-        this.performanceClient.addQueueMeasurement(
-            PerformanceEvents.SilentHandlerInitiateAuthRequest,
-            this.authCodeRequest.correlationId
-        );
-
-        if (!requestUrl) {
-            // Throw error if request URL is empty.
-            this.logger.info("Navigate url is empty");
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.emptyNavigateUri
+/**
+ * Monitors an iframe content window until it loads a url with a known hash, or hits a specified timeout.
+ * @param iframe
+ * @param timeout
+ */
+export async function monitorIframeForHash(
+    iframe: HTMLIFrameElement,
+    timeout: number,
+    pollIntervalMilliseconds: number,
+    performanceClient: IPerformanceClient,
+    logger: Logger,
+    correlationId: string
+): Promise<string> {
```

Comment:
If we end up passing `logger`, `correlationId` and `performanceClient` to all functions (I am hoping more of these are slated as you dig through each flow), does it make sense to create a single data structure when we initiate a flow as `RequestMetaData` or something similar to flow through instead of passing them separately?

Reply:
I've been thinking about potentially having the invoke/invokeAsync functions auto-inject these into each function call. Less to pass in manually and keeps the ordering consistent, however, they would still need to be defined in the function definitions. Unfortunately grouping all of these in an object prevents them from being minified, putting us back to where we are today in terms of size.

Still mulling that over but if you have ideas/suggestions would love to discuss

Reply:
How much is the minification cost if it is one instance that flows across a given flow? We do not intend to create new instances of these unlike classes everytime, just at the beginning until we close a given request?

Reply:
It's not a cost per instance it's a cost per reference

Reply:
If we reference a locally scoped `logger` let's say, each reference can be minified to `a`. But if we nest that inside an object `requestMetadata`, to take your example, each reference to `requestMetadata.logger` can only be minified to `a.logger`